extensions/interactions/ItemSelectionInput/directives/item-selection-input-validation.service.ts

```diff
@@ -208,32 +232,32 @@ export class ItemSelectionInputValidationService {
                 'argument choices.')
             });
           }
+        });
 
-          if (rule.type === 'IsProperSubsetOf') {
-            if (ruleInputs.length < 2) {
-              warningsList.push({
-                type: AppConstants.WARNING_TYPES.ERROR,
-                message: (
-                  'In Oppia response ' + (answerIndex + 1) + ', ' +
-                  'learner answer ' + (ruleIndex + 1) + ', the "proper ' +
-                  'subset" learner answer must include at least 2 options.')
-              });
-            }
-          } else if (rule.type === 'Equals') {
-            if (minAllowedCount > ruleInputs.length ||
-              maxAllowedCount < ruleInputs.length) {
-              warningsList.push({
-                type: AppConstants.WARNING_TYPES.ERROR,
-                message: (
-                  'In Oppia response ' + (answerIndex + 1) + ', ' +
-                  'Learner answer ' + (ruleIndex + 1) + ', the number of ' +
-                  'correct options in the "Equals" learner answer should be ' +
-                  'between ' + minAllowedCount + ' and ' + maxAllowedCount +
-                  ' (the minimum and maximum allowed selection counts).')
-              });
-            }
+        if (rule.type === 'IsProperSubsetOf') {
```

Comment:
Yes, it is. It doesn't belong inside the loop, because it doesn't depend on each ruleInput. Currently it checks some logic about the *rule* but not each ruleInput